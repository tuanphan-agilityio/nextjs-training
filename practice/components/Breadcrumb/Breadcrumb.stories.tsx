import { Meta, Story } from '@storybook/react';
import Breadcrumb, { BreadcrumbProps } from '.';

export default {
  title: 'Breadcrumb',
  component: Breadcrumb,
} as Meta;

const Template: Story<BreadcrumbProps> = (args) => <Breadcrumb {...args} />;

export const Default = Template.bind({});
Default.args = {
  items: [
    { label: 'Home', href: '/' },
    { label: 'Products', href: '/products' },
    { label: 'Category 1', href: '/products/category1' },
  ],
  className: 'text-quaternary',
};

export const CustomStyle = Template.bind({});
CustomStyle.args = {
  items: [
    { label: 'Custom', href: '/custom' },
    { label: 'Design', href: '/custom/design' },
  ],
  className: 'bg-primary text-secondary',
};

export const SingleItem = Template.bind({});
SingleItem.args = {
  items: [{ label: 'Single Item' }],
  className: 'text-quaternary',
};
