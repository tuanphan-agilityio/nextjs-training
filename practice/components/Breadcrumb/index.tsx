import { FC, memo } from 'react';
import Link from 'next/link';
import clsx from 'clsx';

interface BreadcrumbItem {
  label: string;
  href?: string;
}

interface BreadcrumbProps {
  items: BreadcrumbItem[];
  className?: string;
}

const Breadcrumb: FC<BreadcrumbProps> = ({ items, className }) => {
  const isLastItem = (index: number) => index === items.length - 1;

  return (
    <nav aria-label='breadcrumb'>
      <ol className={clsx('flex w-fit text-sm text-primary', className)}>
        {items.map((item, index) => (
          <li
            key={index}
            className={clsx({
              'font-primary-bold': isLastItem(index),
            })}
          >
            {isLastItem(index) ? (
              item.label
            ) : (
              <>
                <Link href={item?.href ?? ''}>
                  <a className='hover:underline'>{item.label}</a>
                </Link>
                <span className='mx-6'>/</span>
              </>
            )}
          </li>
        ))}
      </ol>
    </nav>
  );
};

export type { BreadcrumbProps, BreadcrumbItem };

export default memo(Breadcrumb);
