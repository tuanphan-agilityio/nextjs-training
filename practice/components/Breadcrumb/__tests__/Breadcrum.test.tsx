import { render, screen } from '@testing-library/react';
import Breadcrumb, { BreadcrumbProps } from '..';

describe('Breadcrumb Component', () => {
  const mockProps: BreadcrumbProps = {
    items: [
      { label: 'Home', href: '/' },
      { label: 'Products', href: '/products' },
      { label: 'Category 1', href: '/products/category1' },
    ],
  };

  const renderBreadcrumb = () => render(<Breadcrumb {...mockProps} />);

  it('renders Breadcrumb component with items', () => {
    renderBreadcrumb();

    const breadcrumbItems = screen.getAllByRole('listitem');
    expect(breadcrumbItems).toHaveLength(mockProps.items.length);

    const lastBreadcrumbItem = breadcrumbItems[breadcrumbItems.length - 1];
    expect(lastBreadcrumbItem).toHaveClass('font-primary-bold');
  });

  it('matches snapshot', () => {
    const { container } = renderBreadcrumb();
    expect(container).toMatchSnapshot();
  });
});
