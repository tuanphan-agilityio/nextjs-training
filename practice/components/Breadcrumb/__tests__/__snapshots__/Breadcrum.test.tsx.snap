// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Component matches snapshot 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
  >
    <ol
      class="flex w-fit text-sm text-primary"
    >
      <li
        class=""
      >
        <a
          class="hover:underline"
        >
          Home
        </a>
        <span
          class="mx-6"
        >
          /
        </span>
      </li>
      <li
        class=""
      >
        <a
          class="hover:underline"
        >
          Products
        </a>
        <span
          class="mx-6"
        >
          /
        </span>
      </li>
      <li
        class="font-primary-bold"
      >
        Category 1
      </li>
    </ol>
  </nav>
</div>
`;
